ext/oci8/oci8.stub.php

```diff
@@ -0,0 +1,734 @@
+
+/**
+ * @param resource $statement_resource
+ * @param mixed $column_number_or_name
+ * @return string|int|false|null @todo return false, null?
+ */
+function oci_field_type($statement_resource, $column_number_or_name) {}
+
+/**
+ * @param resource $statement_resource
+ * @param mixed $column_number_or_name
+ * @return int|false|null @todo return false, null?
+ */
+function oci_field_type_raw($statement_resource, $column_number_or_name) {}
+
+/**
+ * @param resource $statement_resource
+ * @param mixed $column_number_or_name
+ * @return bool
+ */
+function oci_field_is_null($statement_resource, $column_number_or_name): bool {}
+
+/**
+ * @todo do we need this?
+ * @param bool $mode
+ * @return void
+ */
+function oci_internal_debug(bool $mode): void {}
+
+/**
+ * @param resource $statement_resource
+ * @param int $mode
+ * @return bool
+ */
+function oci_execute($statement_resource, int $mode = OCI_COMMIT_ON_SUCCESS): bool {}
+
+/**
+ * @param resource $statement_resource
+ * @return bool
+ */
+function oci_cancel($statement_resource): bool {}
+
+/**
+ * @param resource $statement_resource
+ * @return bool
+ */
+function oci_fetch($statement_resource): bool {}
+
+/**
+ * @param resource $statement_resource
+ * @param mixed $result
+ * @param int $mode
+ * @return int|false @todo return false, double check?
+ */
+function ocifetchinto($statement_resource, &$result, int $mode = 0) {}
+
+/**
+ * @param resource $statement_resource
+ * @param mixed $output
+ * @param int $skip
+ * @param int $maximum_rows
+ * @param int $flags
+ * @return int @todo double check.
+ */
+function oci_fetch_all($statement_resource, &$output, int $skip = 0, int $maximum_rows = -1, int $flags = 0): int {}
+
+/**
+ * @param resource $statement_resource
+ * @param int $mode
+ * @return object
+ */
+function oci_fetch_object($statement_resource, int $mode = 0): object {}
+
+/**
+ * @param resource $statement_resource
+ * @return array|false @todo double check.
+ */
+function oci_fetch_row($statement_resource) {}
+
+/**
+ * @param resource $statement_resource
+ * @return array|false @todo double check.
+ */
+function oci_fetch_assoc($statement_resource) {}
+
+/**
+ * @param resource $statement_resource
+ * @param int $mode
+ * @return array|false @todo double check.
+ */
+function oci_fetch_array($statement_resource, int $mode = 0) {}
+
+/**
+ * @param resource $statement_resource
+ * @return bool
+ */
+function oci_free_statement($statement_resource): bool {}
+
+/**
+ * @param resource $connection_resource
+ * @return bool
+ */
+function oci_close($connection_resource): bool {}
+
+/**
+ * @param string $username
+ * @param string $password
+ * @param string|null $connection_string
+ * @param string|null $character_set
+ * @param int $session_mode
+ * @return resource|false @todo double check
+ */
+function oci_new_connect(string $username, string $password, ?string $connection_string = null, ?string $character_set = null, int $session_mode = OCI_DEFAULT) {}
```

Comment:
`$connection_string` and `$character_set` are not nullable

Reply:
@kocsismate Do you happen to know the default value for these parameters?

Reply:
I believe both are `UNKNOWN`. :)

Off-topic: it's weird for me that the `$connection_string` parameter name refers to the `dbname` C variable. I see that it's also a preexisting stuff, but the parameter name is a bit deceptive.

Reply:
I removed nullability and made the default value `UNKNOWN`. Thanks!